operations/restrictions.md

```diff
@@ -369,3 +369,111 @@ Removes restrictions from the service.
 ```javascript
 {}
 ```
+
+## Set restrictions
+
+Adds new restrictions with the specified conditions.
+
+If there already exists a restriction with the same conditions, following scenarios apply:
+
+1) If the exceptions of the new restriction match the old restriction:
+   1) If the new interval is longer than the old one, new restriction is created joining the two intervals.
+   2) If the new interval is shorter, no changes are made.
+2) If the exceptions of the new restriction do not match the old restriction:
+   1) If the new interval is overlaps the old interval, the old restriction will be spliced before and after the new interval. Restrictions matching the old restriction are then added at the appropriate interval along with the new restriction.
+   2) If the new interval do not overlap the old interval, the new restriction is added as usual.
+
+If the supplied restrictions match in all the properties but differ in interval and follow each other chronologically, the supplied restrictions will be joined into a single restriction.
+
+The usage of this endpoint must be enabled per enterprise.
+
+### Request
+
+`[PlatformAddress]/api/connector/v1/restrictions/set`
```

Comment:
Shouldn't we also deprecate the restrictions/set?

Reply:
Shouldn't we deprecate it once partners are happy with the set ?

Reply:
I'm not quite sure of the situation here, but depending on the situation it could be handled in different ways:
1. If the new operation is intended to **sit alongside** 'add restrictions', then they should have separate use cases and separate names, and the names should reflect the use cases; 'set restrictions' and 'add restrictions' sound like the same thing to me.
2. If the new operation **replaces** 'add restrictions' then it could either be a version 2 of the same operation, or it could have a different name, but either way the old operation is **deprecated** - it should be marked as deprecated and listed on the Deprecations page.
3. If the new operation is being trialled and is subject to change, like a Beta test, then it should either be left out of the documentation, or marked as **restricted** with a note to say what the restriction is, e.g. "beta test, subject to change". This is true regardless of whether 1 or 2 above also applies.

Reply:
ok, i'll add #3 and then once we are ready for release, I'll remove it and marked the add as deprecated.
Thanks

Reply:
@MikeAdamsMews fair. I think we should go with 2+3. Introduction of a new endpoint, that replaces the functionality of the restrictions/add. It comes with breaking changes to the 'add' so we were not able to implement it in a non-breaking way. We want to document it for partners but at the same time would like to be able to do some breaking changes if some problems with the endpoint are found. Restricted is a way to go.